### ISSUE TITLE
PickerList - iconFunction

### ISSUE THREAD

**Issue by teotigraphix** [BowlerHatLLC/feathers#1466]
I found myself today wanting a full icon picker list, no text.

In my dataProvider I have icon images which work great for the list, wouldn't rounding out this component with an iconFunction make sense?

I set a labelFunction to return null, but wanted the iconFunction to return the selectedItem in the dataProvider.

So it would be like on change, if an iconFunction is set, replace the current icon with the iconfFunction icon.

I also had to make a custom buttonFactory, just to be able to set the icon, which is fine but this could make this type of implementation very easy.

Am I missing something?

**Comment by joshtynjala** [comment]
I opened #1013 with the same idea a while back, but I ultimately decided not to implement it. The reason is that the icon is typically used for some kind of arrow or something to display that the PickerList can be opened. In some cases, this could be made a part of the button's background skin, but often, being a part of the skin isn't very flexible because the arrow will need to be repositioned depending on the button's dimensions.

**Comment by teotigraphix** [comment]
Ah ok, I am not using the arrow, its a button that popups up the list. So the icon is always the current selection in the list.

**Comment by joshtynjala** [comment]
I understand. The iconField/iconFunction would definitely be helpful for that use-case. I'm a little wary to add it because I think it will be unexpected for the drop-down arrow to "disappear" when someone first tries to use it. I think it's more likely that someone who wants an icon will want a layout closer to the default item renderers, where it supports an extra "accessory". Perhaps that's a change I should consider for the next version because I'd like that to be possible first.

**Comment by teotigraphix** [comment]
Yeah maybe that is what I am asking for, that the button by API be configurable without having to use a buttonFactory.
